string_utils.test.js

```diff
@@ -0,0 +1,66 @@
+/*jslint node: true */
+"use strict";
+
+var StringUtils = require('./string_utils');
+var STRING_JOIN_CHAR = StringUtils.STRING_JOIN_CHAR;
+var getSourceString = StringUtils.getSourceString;
+
+function runTest(name, expectation, func) {
+    var resultMessage = name;
+    var result = func();
+
+    if (result === expectation) {
+        resultMessage += ': PASSED';
+    } else {
+        resultMessage += ': FAILED - "'+result+'" did not equal "'+expectation+'"';
+    }
+
+    console.log(resultMessage);
+}
+
+/**
+ * getSourceString
+ */
+
+var simpleString = 'simple test string';
```

Comment:
as you care more about the data type than the data itself, it might be good to use generators for this if/when #42 goes in

Reply:
I ended up leaving this as-is for now, but happy to go with the string gen instead, I just couldn't get it on the (quick) first go around.